### ISSUE TITLE
Site url wrongly defined; Error occurred while rendering the component 404

### ISSUE THREAD

**Issue by Xilonz** [soflyy/oxygen-bugs-and-features#886]
**Describe the bug**
In component-init.php line 1030 the site url is wrongly defined, causing 404 errors in the editor when WordPress is in a different directory than the site home, more specifically, when the WordPress Address is not the same as the site address. It breaks the editor.

**To Reproduce**

1. Install WordPress in a subdirectory eg. /wp/ (bedrock might be easier for this example)
2. Set the home url to the root domain

**Expected behavior**
No 404's 

**What actually happens**
404's

**Screenshots**
![Capture](https://user-images.githubusercontent.com/15018940/76225986-839e8880-621d-11ea-9191-16e3f1b8da2f.PNG)

**Desktop (please complete the following information):**
 - Oxygen Version 3.2

**Additional context**
Instead of using [`get_site_url()`](https://developer.wordpress.org/reference/functions/get_site_url/) in component-init.php use [`get_home_url()`](https://developer.wordpress.org/reference/functions/get_home_url/) on line 1030

I've verified this solves the issue.
Edit; This issue returns in multiple files, please evaluate all uses of get_site_url() in the code.

**Comment by robrecord** [comment]
I believe I'm getting this on our Bedrock-enabled site. Can you fix it please? Causing us a variety of problems. Cheers!

**Comment by robrecord** [comment]
Could someone please let me know why, after my editing the aforementioned line, the javascript does not use this updated value on the front end? I can see it coming through successfully in the HTML, in an options object inside a <script> tag. Is there a cache I must purge? I have emptied browser cache, cookies and local storage.

**Comment by robrecord** [comment]
To reproduce this, set up a site using [Bedrock](https://roots.io/bedrock/), add the oxygen plugin, edit the Archive template and set Previewing to Posts.

**Comment by robrecord** [comment]
It would appear that if the "Home" and "Posts" pages are set to specific pages, the aforementioned fixes are not enough to specify the correct URL to preview the template for Posts. Perhaps further code is required to get the right URL for the posts page.

**Comment by robrecord** [comment]
This is the fix for the above:

Replace line 1215 in oxygen/component-framework/includes/templates.php with:
```php
$permalink = get_option( 'page_for_posts' ) ? get_page_link( get_option( 'page_for_posts' ) ) : get_home_url();
```

**Comment by jason-schell** [comment]
I have the same issue on all of my projects. Still waiting for a bugfix. Right now I changed the WordPress-URL back to the same as root.

**Comment by KittenCodes** [comment]
Fixed in v3.8 Beta 1.
